### PR TITLE
Update CloudStack immutable fields docs

### LINKED ISSUE #5312: Update upgrade cluster doc and add cloudstack fields in the "Upgradeable Cluster Attributes" section
Add upgradeable cluster Attributes section for Cloudstack cluster components.  
https://anywhere.eks.amazonaws.com/docs/tasks/cluster/cluster-upgrades/vsphere-and-cloudstack-upgrades/#upgradeable-attributes

### PR DESCRIPTION
**Pull request by chrisdoherty4**
## Summary

Add documentation on CloudStack immutable fields.

Closes #5312

### CODE REVIEW

File: docs/content/en/docs/tasks/cluster/cluster-upgrades/vsphere-and-cloudstack-upgrades.md [hunk 1]
```diff
@@ -215,6 +215,17 @@ allowing you to upgrade a number of fields simultaneously with the same procedur
 - `osFamily`
 - `network`
 
+`CloudStackDatacenterConfig`:
+- `availabilityZones` (Can add and remove availability zones provided at least 1 previously configured zone is still present)
+
+`CloudStackMachineConfig`:
+- `template`
+- `computeOffering`
+- `diskOffering`
```
Comment (ddjjia) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades/vsphere-and-cloudstack-upgrades.md:224]: I remember diskOffering should be mutable once we consume capi 1.4? 
https://github.com/kubernetes-sigs/cluster-api/blob/v1.4.0/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go#L172-L175
Or do you plan to remove it once we actually consume 1.4?

Comment (chrisdoherty4) [reply docs/content/en/docs/tasks/cluster/cluster-upgrades/vsphere-and-cloudstack-upgrades.md:224]: [Its already mutable.](https://github.com/aws/eks-anywhere/blob/a3e160645c4e0a9bff3350870131d3bb2fa9fbec/pkg/providers/cloudstack/cloudstack.go#L522)

**Pull request merged**